### ISSUE TITLE
Parse "^" causes segmentation fault?

### ISSUE THREAD

**Issue by Marlin-Na** [symengine/symengine#1351]
It seems that using  `basic_parse` to parse "3 ^ 2" would cause segmentation fault, while "3 ** 2" works as expected.

For example using the following code:

```c
#include <symengine/cwrapper.h>
#include <stdio.h>

int main () {
    basic x;
    basic_new_stack(x);
    CWRAPPER_OUTPUT_TYPE o = basic_parse(x, "3 ^ 2");
    basic_free_stack(x);
    printf("cwrapper output type: %d \n", o);
    return 0;
}
```

Version of SymEngine is 0.3.0, I am not sure whether it is fixed in newer version.

**Comment by ShikharJ** [comment]
@Marlin-Na We recently updated the parser to provide the usage for `^` as `logical_xor` which expects `SymEngine::Boolean` type objects. As such, I think the segfault is occurring because the parser is trying to `rcp_static_cast<const Boolean>()` the two arguments. Additionally, you can look [here](https://github.com/symengine/symengine/pull/1298).

**Comment by isuruf** [comment]
@ShikharJ, I think we shouldn't use `^` for xor by default. Even sympy converts `^` to power by default and there is a `convert_xor` parameter for switching.

**Comment by ShikharJ** [comment]
@isuruf I am getting this on `SymPy 1.1.1`:
```
In [1]: 3 ^ 2
Out[1]: 1

In [2]: 0^1
Out[2]: 1

In [3]: 1^1
Out[3]: 0

In [4]: x ^ y
Out[4]: x ⊻ y

In [5]: type(x ^ y)
Out[5]: Xor
```

**Comment by isuruf** [comment]
@ShikharJ, I was talking about sympy's parser.
```
In [6]: S("x ^ y")
Out[6]: x**y
```

**Comment by ShikharJ** [comment]
Closed via #1352.
